game-core/src/main/java/tools/image/AutoPlacementFinder.java

```diff
@@ -260,56 +260,44 @@ private void calculate() {
-        final Set<Polygon> containedPolygons = mapData.getContainedTerritoryPolygons(name);
-        final List<Point> points =
-            containedPolygons.isEmpty()
-                ? getPlacementsStartingAtMiddle(
-                    mapData.getPolygons(name),
-                    mapData.getBoundingRect(name),
-                    mapData.getCenter(name))
-                : getPlacementsStartingAtTopLeft(
-                    mapData.getPolygons(name),
-                    mapData.getBoundingRect(name),
-                    mapData.getCenter(name),
-                    containedPolygons);
-        placements.put(name, points);
-        textOptionPane.appendNewLine(name + ": " + points.size());
-      }
-      textOptionPane.appendNewLine("\r\nAll Finished!");
-      textOptionPane.countDown();
-      final String fileName =
-          new FileSave("Where To Save place.txt ?", "place.txt", mapFolderLocation).getPathString();
-      if (fileName == null) {
-        textOptionPane.appendNewLine("You chose not to save, Shutting down");
-        textOptionPane.dispose();
-        return;
-      }
-      try (OutputStream os = new FileOutputStream(fileName)) {
-        PointFileReaderWriter.writeOneToMany(os, placements);
-        textOptionPane.appendNewLine("Data written to :" + new File(fileName).getCanonicalPath());
-      } catch (final IOException e) {
-        log.log(Level.SEVERE, "Failed to write points file: " + fileName, e);
-        textOptionPane.dispose();
-        return;
-      }
+    MapData mapData = new MapData(mapDir);
+    textOptionPane.show();
+    textOptionPane.appendNewLine(
+        "Place Dimensions in pixels, being used: " + placeWidth + "x" + placeHeight + "\r\n");
+    textOptionPane.appendNewLine("Calculating, this may take a while...\r\n");
+    final Map<String, List<Point>> placements = new HashMap<>();
+    for (final String name : mapData.getTerritories()) {
+      final Set<Polygon> containedPolygons = mapData.getContainedTerritoryPolygons(name);
+      final List<Point> points =
+          containedPolygons.isEmpty()
+              ? getPlacementsStartingAtMiddle(
+                  mapData.getPolygons(name), mapData.getBoundingRect(name), mapData.getCenter(name))
+              : getPlacementsStartingAtTopLeft(
+                  mapData.getPolygons(name),
+                  mapData.getBoundingRect(name),
+                  mapData.getCenter(name),
+                  containedPolygons);
+      placements.put(name, points);
+      textOptionPane.appendNewLine(name + ": " + points.size());
+    }
+    textOptionPane.appendNewLine("\r\nAll Finished!");
+    textOptionPane.countDown();
+    final String fileName =
+        new FileSave("Where To Save place.txt ?", "place.txt", mapFolderLocation).getPathString();
+    if (fileName == null) {
+      textOptionPane.appendNewLine("You chose not to save, Shutting down");
       textOptionPane.dispose();
-    } catch (final Exception e) {
```

Comment:
This catch block would never have been triggered from `new MapData` to begin with, and is removed.

Reply:
Not sure about that either.
This catches any potential RuntimeExceptions as well and loadImage does still explicitly throw one, so the constructor could definitely throw an exception that could land here.

Reply:
I'm less sure now as well. Though, looking at the error handling, we would have shown a user 4 pop-ups, and none of which were very specific. 

Considering this is in the map-making `tools`, I think letting it throw is probably okay.  @RoiEXLab WDYT?